tests/e2e/cucumber/steps/ui/resources.ts

```diff
@@ -258,6 +261,31 @@ When(
   }
 )
 
+export const processDelete = async (stepTable: DataTable, pageObject: any, actionType: string) => {
+  let files, parentFolder
+  const deleteInfo = stepTable.hashes().reduce((acc, stepRow) => {
+    const { resource, from } = stepRow
+    const resourceInfo = {
+      name: resource
+    }
+    if (!acc[from]) {
+      acc[from] = []
+    }
+    acc[from].push(resourceInfo)
+    return acc
+  }, {})
+
+  for (const folder of Object.keys(deleteInfo)) {
+    files = deleteInfo[folder]
+    parentFolder = folder !== 'undefined' ? folder : null
+    await pageObject.delete({
+      folder: parentFolder,
+      resourcesWithInfo: files,
+      via: actionType === 'batch action' ? 'BATCH_ACTION' : 'SIDEBAR_PANEL'
+    })
+  }
+}
```

Comment:
We can implement it in step def and the `deleteResource` function in `resource/actions.ts`. And we might not need this new function.

Reply:
this function is used in other steps to so keeping it.